wagtail/wagtailforms/templates/wagtailforms/list_submissions.html

```diff
@@ -6,25 +6,29 @@
     <col />
     <thead>
         <tr>
+            <th colspan="{{ data_headings|length|add:1 }}">
+                <button class="button no" id="delete-submissions" style="visibility: hidden">Delete selected submissions</button>
+            </th>
+        </tr>
+        <tr>
+            <th><input type="checkbox" id="select-all" /></th>
             {% for heading in data_headings %}
                 <th>{{ heading }}</th>
             {% endfor %}
-            <th>{% trans "Actions" %}</th>
         </tr>
     </thead>
     <tbody>
         {% for row in data_rows %}
             <tr>
+                <td>
+                    <input type="checkbox" name="select-{{ row.model_id }}" class="select-submission" data-submission-id="{{ row.model_id }}" />
```

Comment:
The checkbox handling could be cleaned up a lot here by giving all checkboxes the same name and a `value` equal to the submission ID:

    <input type="checkbox" name="submission_id" value="{{ row.model_id }}">

This will then submit as something like `submission_id=1&submission_id=2&submission_id=3` and the list of IDs can be extracted with `request.GET.getlist('submission_id')`, avoiding the need for the string munging to remove the `select-` prefix.